runtime/v2/manager.go

```diff
@@ -140,14 +141,73 @@ func (m *MM) ValidateGenesis(genesisData map[string]json.RawMessage) error {
 	return nil
 }
 
-// InitGenesis performs init genesis functionality for modules.
-func (m *MM) InitGenesis() {
-	panic("implement me")
+// ExportGenesis performs export genesis functionality for modules
+func (m *MM) ExportGenesis(ctx context.Context) (map[string]json.RawMessage, error) {
+	return m.ExportGenesisForModules(ctx, []string{})
 }
 
-// ExportGenesis performs export genesis functionality for modules
-func (m *MM) ExportGenesis() {
-	panic("implement me")
+// ExportGenesisForModules performs export genesis functionality for modules
+func (m *MM) ExportGenesisForModules(ctx context.Context, modulesToExport []string) (map[string]json.RawMessage, error) {
+	if len(modulesToExport) == 0 {
+		modulesToExport = m.config.ExportGenesis
+	}
+	// verify modules exists in app, so that we don't panic in the middle of an export
+	if err := m.checkModulesExists(modulesToExport); err != nil {
+		return nil, err
+	}
+
+	type genesisResult struct {
+		bz  json.RawMessage
+		err error
+	}
+
+	channels := make(map[string]chan genesisResult)
+	for _, moduleName := range modulesToExport {
+		mod := m.modules[moduleName]
+		if module, ok := mod.(appmodulev2.HasGenesis); ok {
+			channels[moduleName] = make(chan genesisResult)
+			go func(module appmodulev2.HasGenesis, ch chan genesisResult) {
+				jm, err := module.ExportGenesis(ctx)
+				if err != nil {
+					ch <- genesisResult{nil, err}
+					return
+				}
+				ch <- genesisResult{jm, nil}
+			}(module, channels[moduleName])
+		} else if module, ok := mod.(sdkmodule.HasABCIGenesis); ok {
+			channels[moduleName] = make(chan genesisResult)
+			go func(module sdkmodule.HasABCIGenesis, ch chan genesisResult) {
+				jm, err := module.ExportGenesis(ctx)
+				if err != nil {
+					ch <- genesisResult{nil, err}
+				}
+				ch <- genesisResult{jm, nil}
+			}(module, channels[moduleName])
+		}
+	}
```

Comment:
Modules that do not implement the export method are silently ignored. Would some log output make sense?

Reply:
All mandatory modules are present and checked by `assertNoForgottenModules`.
I'd rather have `assertNoForgottenModules` panic when unnecessary modules are present than adding extra log personally.

We have relaxed the required modules in the previous sdk release, but you could still add modules to say `OrderBeginBlock` that don't implement BeginBlocker, and it will be silently skipped. Given this is a green field, imho best to `assertNoForgottenModules` more pedantic in this new module manager.